src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/endpoints/StreamedEndpoint.kt

```diff
@@ -1,16 +1,36 @@
 package org.vaccineimpact.api.app.controllers.endpoints
 
+import org.vaccineimpact.api.app.app_start.Endpoint
 import org.vaccineimpact.api.app.context.ActionContext
 import org.vaccineimpact.api.serialization.MontaguSerializer
 import org.vaccineimpact.api.serialization.Serializer
 import org.vaccineimpact.api.serialization.StreamSerializable
 
+// Old style endpoints
 fun <TRepository> ((ActionContext, TRepository) -> StreamSerializable<*>).streamed()
         : (ActionContext, TRepository) -> Unit
 {
     return { context, repo -> stream(this(context, repo), context) }
 }
 
+// New style endpoints
+fun Endpoint.streamed(): Endpoint
+{
+    return this.copy(postProcess = ::streamIfStreamable)
+}
+fun streamIfStreamable(data: Any?, context: ActionContext): Any?
+{
+    if (data is StreamSerializable<*>)
+    {
+        return stream(data, context)
+    }
+    else
+    {
+        val typeName = data?.javaClass?.simpleName ?: "unknown"
+        throw Exception("Attempted to stream $data (type: $typeName, but it is not StreamSerializable")
+    }
```

Comment:
I remember this `javaClass.simpleName` property being not very human readable, see this commit and associated ticket: 
https://github.com/vimc/montagu-api/commit/e6a634f9f3ad9618c70935e56aad17d081a3b02a
https://vimc.myjetbrains.com/youtrack/issue/VIMC-744

Reply:
:+1: